aptos-move/framework/aptos-framework/sources/aggregator_table.move

```diff
@@ -0,0 +1,103 @@
+/// This module provides foundations to create aggregators in the system.
+module aptos_framework::aggregator_table {
+    use std::error;
+    use std::signer;
+
+    use aptos_framework::aggregator::{Self, Aggregator};
+    use aptos_framework::system_addresses;
+    use aptos_framework::table::{Self, Table};
+    use aptos_framework::timestamp;
+
+    /// When aggregator table has already been published.
+    const EAGGREGATOR_TABLE_EXISTS: u64 = 0x1;
+
+    /// Struct that stores aggregators, as a (key, value) pair, where `key` is
+    /// a unique key that can be used to identify an aggregator, and `value` is
+    /// an actual value.
+    /// Note that the **only** way to access the value is via `Aggregator` API.
+    struct AggregatorTable has key {
```

Comment:
How is this aggregator table going to be used?

Reply:
In short: we will place it on the core account so that no one else can create new aggregators. When we need aggregator, we query this "table" and create a new aggregator that the user can manipulate. Once all updates, etc. are done, the user destroys the aggregator. For example, if balance was an `Aggregator`, then when we initialize the `CoinStore` we would initialise balance field as aggregator.